### DISCUSSION TITLE
Update interval

### DISCUSSION THREAD

**Discussion by clee604** [KrystianD/bm2-battery-monitor#6] [Q&A] [answered]
Hi there,

Big thanks for this, I have it setup using a nodemcu and it works well to monitor and send voltage data to Home Assistant using MQTT from a 12v marine battery connected to some solar panels.

It seems to update super frequently (ie. every second or so), I'm wondering if it is possible to set or reduce the frequency it updates and polls? I'm not sure if it would save on the battery usage of the nodemcu at all but seems overkill to be updating so often in my use case.

Thanks!

**Comment by KrystianD** [comment, marked as answer]
Hi, I am really glad you like it!

As of now, I do not see any way of reducing update interval. It is the module which sends updates (not nodemcu that asks for them) via BLE Notifications.

I took a quick look at the code, but don't see any way to reduce notifications interval. You can try to find that in their app. Let me know if you find it, then we can think of implementing it.

**Comment by clee604** [comment]
That explains it well, thanks!

I was trying to make changes to the config within HA with no results and now I know why.

On a side (probably unrelated) note, I notice this error in the ESPHome add-on when I go into the logs now:

```
INFO Reading configuration /config/esphome/bm2-nodemcu.yaml...
Failed config
sensor.ble_client: [source /config/esphome/bm2.yaml:8]
  Looks like you're trying to create a ble characteristic sensor. Please add `type: characteristic` to your sensor config.
```

Not sure if I messed something up when I was tinkering with the configuration or if some new update (maybe the new bluetooth update in HA?) caused the error. The voltage sensor is still working fine though in my dashboard.

**Reply by KrystianD** [reply]
Looks like they added this `type` parameter. Can you try to add this to `bm2.yaml` file and check if it works?

**Reply by clee604** [reply]
adding `type: characteristic` fixed it. I thought i needed to put an actual characteristic and not the word 'characteristic'

Seems kinda redundant if the error already knows what the problem is but anyways you're right, it's all good now.

**Reply by KrystianD** [reply]
okay, good to know.

> Seems kinda redundant

Looks like they introduced new type, which is `type: rssi` with the default fallback to `characteristic` for backwards compatibility, which is the right way of doing this :)
